trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1138,4 +1170,94 @@ private static Response validateParameters(String outputId, QueryParameters quer
+     *
+     * @param expUUID
+     *            desired experiment UUID
+     * @param outputId
+     *            Output ID for the data provider to query
+     * @param typeId
+     *            the configuration source type ID
+     * @return a configuration source type
+     */
+    @GET
+    @Path("/{outputId}/configTypes/{typeId}")
+    @Tag(name = OCG)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Get a single configuration source type defined on the server for a given data provider and experiment.", responses = {
+            @ApiResponse(responseCode = "200", description = "Returns a single configuration source type", content = @Content(schema = @Schema(implementation = org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model.ConfigurationSourceType.class))),
+            @ApiResponse(responseCode = "404", description = PROVIDER_NOT_FOUND, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = NO_SUCH_CONFIGURATION, content = @Content(schema = @Schema(implementation = String.class))),
+    })
+    public Response getConfigurationType(
+            @Parameter(description = EXP_UUID) @PathParam("expUUID") UUID expUUID,
+            @Parameter(description = OUTPUT_ID) @PathParam("outputId") String outputId,
+            @Parameter(description = CFG_TYPE_ID) @PathParam("typeId") String typeId) {
+        return Response.status(Status.NOT_IMPLEMENTED).entity("Get a configuration type for a given output").build(); //$NON-NLS-1$
+    }
+
+    /**
+     * POST a custom configuration to the data provider to create derived data
+     * providers.
+     *
+     * @param expUUID
+     *            desired experiment UUID
+     * @param outputId
+     *            Output ID for the data provider to query
+     * @param typeId
+     *            the configuration source type ID
+     * @param queryParameters
+     *            the query parameters used to create a data provider
+     * @return a list of data provider descriptors
+     */
+    @POST
+    @Path("/{outputId}")
+    @Tag(name = OCG)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Get the list of outputs for this configuration", responses = {
+            @ApiResponse(responseCode = "200", description = "Returns a list of output provider descriptors", content = @Content(array = @ArraySchema(schema = @Schema(implementation = DataProvider.class)))),
+            @ApiResponse(responseCode = "400", description = INVALID_PARAMETERS, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = PROVIDER_NOT_FOUND, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = NO_SUCH_CONFIGURATION, content = @Content(schema = @Schema(implementation = String.class))),
```

Comment:
Can this happen? Is it when the typeId inside the OutputConfigurationQueryParameters is not recognized by the data provider (outputId)?

Reply:
Yes, if configType for a typeId doesn't exists for the given output.